src/main.js

```diff
@@ -95,6 +96,26 @@ function getNLSConfiguration() {
 	return { locale: initialLocale, availableLanguages: {} };
 }
 
+function migrateUserDataDir(newDir) {
+	var oldDir = path.join(process.env.HOME, '.config', product.nameShort);
+	if (!fs.existsSync(oldDir)) {
+		return;
+	}
+	mkdirp(path.dirname(newDir));
+	fs.renameSync(oldDir, newDir);
+}
+
+function mkdirp(newDir) {
+	var stack = [];
+	while (!fs.existsSync(newDir)) {
+		stack.push(newDir);
+		newDir = path.dirname(newDir);
+	}
+	while (stack.length > 0) {
+		fs.mkdirSync(stack.pop());
+	}
+}
+
```

Comment:
Note that any fs operation in the startup phase here can cause an error and this would be incredibly hard to find out and debug without error handling and fallbacks...